pallets/connectors/src/routers.rs

```diff
@@ -22,9 +22,32 @@ pub struct XcmDomain<CurrencyId> {
 	/// It should contain the pallet index + the `transact` call index, to which
 	/// we will append the eth_tx param. You can obtain this value by building
 	/// an ethereum_xcm::transact call with Polkadot JS on the target chain.
-	pub ethereum_xcm_transact_call_index: Vec<u8>,
+	pub ethereum_xcm_transact_call_index:
+		BoundedVec<u8, ConstU32<{ xcm_primitives::MAX_ETHEREUM_XCM_INPUT_SIZE }>>,
 	/// The ConnectorsXcmRouter contract address on a given domain
 	pub contract_address: H160,
 	/// The currency in which execution fees will be paid on
 	pub fee_currency: CurrencyId,
 }
+
+// NOTE: Remove this custom implementation once the following underlying data implements MaxEncodedLen:
+/// * Polkadot Repo: xcm::VersionedMultiLocation
+/// * PureStake Repo: pallet_xcm_transactor::Config<Self = T>::CurrencyId
+impl<CurrencyId> MaxEncodedLen for XcmDomain<CurrencyId>
+where
+	XcmDomain<CurrencyId>: Encode,
+{
+	fn max_encoded_len() -> usize {
+		// custom MEL bound for `VersionedMultiLocation`
+		xcm::v1::MultiLocation::max_encoded_len()
+			// VersionedMultiLocation is enum with two variants
+			.saturating_add(2)
+			.saturating_add(BoundedVec::<
+				u8,
+				ConstU32<{ xcm_primitives::MAX_ETHEREUM_XCM_INPUT_SIZE }>,
+			>::max_encoded_len())
+			// custom MEL bound for CurrencyId
+			.saturating_add(cfg_types::tokens::CurrencyId::max_encoded_len())
+			.saturating_add(H160::max_encoded_len())
```

Comment:
It would be great to simplify and extend these comments to inform which field of `XcmDomain` they are _add_ing to the `max_encoded_len`:

```suggestion
      // Contract address
			.saturating_add(H160::max_encoded_len())
			// The fee currency 
			.saturating_add(cfg_types::tokens::CurrencyId::max_encoded_len())

```

Moreover, these two types are in the inverse order now, the suggestion left here sorts them. It sums to the same but it's helpful to keep it in order